### DISCUSSION TITLE
sir kl wala code nhi mila

### DISCUSSION THREAD

**Discussion by Saibtain** [AammarTufail/python_ka_chilla2023#82] [General]
kia kisi ko kal wala code mila jo sir na kha tha k ma code send kr doon ga

**Comment by usamabajwa86** [comment]
# libraries and model import
import numpy as np
from sklearn.model_selection import train_test_split, cross_val_score

from sklearn.linear_model import LogisticRegression
from sklearn.neighbors import KNeighborsClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.svm import SVC

# define the models to be evaluated
models = [LogisticRegression(), KNeighborsClassifier(), RandomForestClassifier(), SVC()]
names = ["Logistic Regression", "KNN", "Random Forest", "SVM"]

# perform k-fold cross-validation for each model
k = 10
for name, model in zip(names, models):
    cv_scores = cross_val_score(model, X_train, y_train, cv=k)
    print(f"{name} CV accuracy: {np.mean(cv_scores):.3f} +/- {np.std(cv_scores):.3f}")

**Reply by usamabajwa86** [reply]
from sklearn.model_selection import GridSearchCV

# Define the parameter grid
param_grid = {'C': [0.1, 1, 10],
              'kernel': ['linear', 'rbf'],
              'gamma': [0.1, 1, 10]}

grid_search = GridSearchCV(SVC(), param_grid, cv=5)
grid_search.fit(X_train, y_train)

print("Best parameters: {}".format(grid_search.best_params_))
print("Best cross-validation score: {:.2f}".format(grid_search.best_score_))

**Comment by Muhammad-Ali-Butt** [comment]
@Saibtain bhai agar best model selection walay ki baat kr rai ho jo kal raat kia ha wo mujhay b ni mila github par kahi but maine wo video dobara dekhi aur code saath saath likh lia its only 23 mins long video. 

https://youtu.be/XiUJwXglo5s

**Reply by nouman073** [reply]
![image](https://user-images.githubusercontent.com/76868703/213864011-0747ae90-954d-4e4c-a3d6-02ec1be5a50c.png)

I wrote code from video  but my output showing like that

**Reply by AzamBukhari** [reply]
@nouman073 bhai, ap neechey wali line mai model_name ki juga model_names istemal ker rahey hain

models_scores.append([model_name,accuracy])
